### PR TITLE
Add logger and localization

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: web-extensions/chrome/engine/utils.js [hunk 1]
```diff
@@ -33,16 +33,14 @@ FrameFinder.findWindow = function(frameDef) {
         // supposed to happen). As such, we expect testing to cover this, and
         // in the worst case want to make it extra clear/visible to a used to
         // report the issue to the developers
-        alert('Cannot find an frame with unknown Definition. Please raise the '
-              + 'issue to the developers, as it is due to a development '
-              + 'mistake.');
+        alert(LOCALIZATION.getString('S35'));
```
Comment (Joacchim) [inline web-extensions/chrome/engine/utils.js]: Here too, refering an Error ID would be useful :)

File: web-extensions/chrome/entrypoint.js [hunk 2]
```diff
@@ -41,9 +41,9 @@ if (window.top === window) {
         const engine = new BmcEngine(window.location.origin,
                                      window.location.hostname,
                                      response.resource.comic);
-        console.log('Instanciated BmcEngine');
+        LOGS.log('E0035');
         engine.setup();
     });
 } else {
-    console.warn(`Attempting to reload for iframe: ${window.location}`);
+    LOGS.warn('E0036', {'location': window.location});
```
Comment (Joacchim) [inline web-extensions/chrome/entrypoint.js]: The log message is expecting "iframe" key instead, here.

File: web-extensions/chrome/sidebar.html [hunk 3]
```diff
@@ -128,6 +128,7 @@
         background-color: #7fec34;
       }
     </style>
+    <script src="strings.js"></script>
```
Comment (Joacchim) [inline web-extensions/chrome/sidebar.html]: The sidebar is including strings.js but it is not used in scripts/load_bookmarks.js ? :)

File: web-extensions/chrome/engine/bookmycomics.js [hunk 4]
```diff
@@ -32,7 +32,8 @@ function BmcEngine(hostOrigin, readerName, comicInfo) {
             this.register(evData.label, err => {
                 let retErr = null;
                 if (err) {
-                    retErr = new Error('Could not register Comic ' + evData.label + ': ' + err.message);
+                    retErr = new Error(LOCALIZATION.getString('S15', {'label': evData.label,
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply web-extensions/chrome/engine/bookmycomics.js]: That's a good point. I don't really know what rule to put... Should I remove error codes from log/debug messages and put some for `Error` types?

[the remark this answers is not in the document]
Comment (DavidPineauScality) [reply web-extensions/chrome/engine/bookmycomics.js]: I personally think we should have error codes for anything relating to "reporting an internal issue, deviating from the happy path".

But this means that I don't think we should have error codes for debug/traces/info logs.

File: web-extensions/chrome/sidebar.html (hunk 3 above)
Comment (GuillaumeGomez) [reply web-extensions/chrome/sidebar.html]: Oh indeed!

**Comment by GuillaumeGomez** [comment]
Updated.

File: web-extensions/chrome/entrypoint.js [hunk 5]
```diff
@@ -41,9 +41,9 @@ if (window.top === window) {
         const engine = new BmcEngine(window.location.origin,
                                      window.location.hostname,
                                      response.resource.comic);
-        console.log('Instanciated BmcEngine');
+        LOGS.log('S42');
         engine.setup();
     });
 } else {
-    console.warn(`Attempting to reload for iframe: ${window.location}`);
+    LOGS.warn('E0003', {'location': window.location});
```
Comment (Joacchim) [inline web-extensions/chrome/entrypoint.js]: the string is still expecting `iframe` in its format :)

Comment (GuillaumeGomez) [reply web-extensions/chrome/entrypoint.js]: Woups. Fixed!

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by Joacchim** [comment]
No tests yet, merging :)

**Pull request merged**